src/models/networkCurrency.model.ts

```diff
@@ -0,0 +1,114 @@
+  /**
+   * Creates a network currency from a network currency DTO
+   * @static
+   * @param {NetworkCurrencyDTO} networkCurrencyDTO
+   * @returns {NetworkCurrency}
+   */
+  public static createFromDTO(networkCurrencyDTO: NetworkCurrencyDTO): NetworkCurrency {
+    return new NetworkCurrency(
+      new NamespaceId(networkCurrencyDTO.namespaceId),
+      networkCurrencyDTO.divisibility,
+    )
+  }
+
+  /**
+   * Creates a NetworkCurrency from the first block transactions
+   * @static
+   * @param {Transaction[]} transactions
+   * @returns {NetworkCurrency}
+   */
+  public static createFromFirstBlockTransactions(transactions: Transaction[]): NetworkCurrency {
+    // read first mosaic definition transaction
+    const mosaicDefinitionTx = transactions.find(
+      (tx) => tx instanceof MosaicDefinitionTransaction,
+    ) as MosaicDefinitionTransaction
+
+    if (mosaicDefinitionTx === undefined) {throw new Error('mosaic definition transaction not found')}
+
+    // read first root namespace
+    const rootNamespaceTx = transactions.find(
+      (tx) => tx instanceof NamespaceRegistrationTransaction
+        && tx.registrationType === NamespaceRegistrationType.RootNamespace,
+    ) as NamespaceRegistrationTransaction
+
+    if (rootNamespaceTx === undefined) {throw new Error('root namespace creation transaction not found')}
+
+    // read sub namespace
+    const subNamespaceTx = transactions.find(
+      (tx) => tx instanceof NamespaceRegistrationTransaction
+        && tx.registrationType === NamespaceRegistrationType.SubNamespace
+        && tx.parentId
+        && tx.parentId.equals(rootNamespaceTx.namespaceId),
+    ) as NamespaceRegistrationTransaction
+
+    if (rootNamespaceTx === undefined) {throw new Error('sub namespace creation transaction was not found')}
+
+    // build network mosaic name
+    const mosaicName = [rootNamespaceTx.namespaceName, subNamespaceTx.namespaceName].join('.')
+
+    return new NetworkCurrency(new NamespaceId(mosaicName), mosaicDefinitionTx.divisibility)
+  }
+
+  /**
+   *Creates an instance of NetworkCurrency.
+   * @param {NamespaceId} namespaceId
+   * @param {number} divisibility
+   */
+  private constructor(
+    public readonly namespaceId: NamespaceId,
+    public readonly divisibility: number,
+  ) {}
+
+  /**
+   * Creates a network currency DTO
+   * @returns {NetworkCurrencyDTO}
+   */
+  public toDTO(): NetworkCurrencyDTO {
+    const id = this.namespaceId.fullName ? this.namespaceId.fullName : this.namespaceId.toHex()
+    return {namespaceId: id, divisibility: this.divisibility}
+  }
+
+  /**
+   * Creates a mosaic given a relative amount
+   * @param {(number | string)} amount
+   * @returns {Mosaic}
+   */
+  public createRelative(amount: number | string): Mosaic {
+    const theAmount = Number(amount)
+    if (theAmount < 0) { throw new Error('The provided amount should be greater than 0') }
+    const absoluteAmount = Math.pow(theAmount, this.divisibility)
```

Comment:
This formula looks wrong (e.g. 10 ^ 6 = 1.000.000 and should be 10.000.000). 

Correct formula:

`const absoluteAmount = amount * Math.pow(10, this.divisibility))`

Reply:
oops